src/subclass/widget.rs

```diff
@@ -353,6 +716,309 @@ where
 //     imp.can_activate_accel(&wrap, signal_id) as glib_sys::gboolean
 // }
 
+unsafe extern "C" fn widget_child_notify<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkWidget,
+    paramptr: *mut gobject_sys::GParamSpec,
+)
+where
+    T: WidgetImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Widget = from_glib_borrow(ptr);
+    let paramwrap: glib::ParamSpec = from_glib_borrow(paramptr);
+
+    imp.child_notify(&wrap, &paramwrap)
+}
+
+unsafe extern "C" fn widget_composited_changed<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkWidget,
+)
+where
+    T: WidgetImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Widget = from_glib_borrow(ptr);
+
+    imp.composited_changed(&wrap)
+}
+
+unsafe extern "C" fn widget_compute_expand<T: ObjectSubclass>(
+    ptr: *mut gtk_sys::GtkWidget,
+    hexpand_ptr: *mut glib_sys::gboolean,
+    vexpand_ptr: *mut glib_sys::gboolean,
+)
+where
+    T: WidgetImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Widget = from_glib_borrow(ptr);
+    let mut hexpand_p: bool = *hexpand_ptr != 0;
+    let mut vexpand_p: bool = *vexpand_ptr != 0;
```

Comment:
Why not `from_glib()` here?

Reply:
`error[E0277]: the trait bound `bool: glib::translate::FromGlib<*mut i32>` is not satisfied`

Reply:
`from_glib(*hexpand_ptr)`, you need to dereference

Reply:
Yup, figured that out and pushed it.